firmware/sys/storage/storage.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2022 Mesh4all <mesh4all.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @brief       storage file
+ *
+ * @author      xkevin190 <kevinvelasco193@gmail.com>
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "shell.h"
+#include "shell_commands.h"
+#include "msg.h"
+#include "system_init.h"
+
+#include "log.h"
+#include "board.h"
+#include "mtd.h"
+#include "mtd_flashpage.h"
+#include "storage.h"
+
+static mtd_dev_t _dev = MTD_FLASHPAGE_INIT_VAL(8);
+static mtd_dev_t *dev = &_dev;
+
+int mtd_start(void) {
+    int ret = mtd_init(dev);
+    if (ret < 0) {
+        printf("Error to the init mtd \n");
+    }
+    return ret;
+}
+
+int mtd_save(uint32_t key, void *value) {
+    uint8_t buf[64];
+
+    memset(buf, 0, sizeof(buf));
+    memcpy(buf, value, sizeof(buf));
+    int ret = mtd_write(dev, buf, key, sizeof(buf));
+
+    if (ret < 0) {
+        printf("Error to the write in mtd %d \n", ret);
+    }
+
+    return ret;
+}
+
+int mtd_read_u8(uint32_t key, uint8_t *output) {
+    uint8_t buf_empty[3];
+    memset(buf_empty, FLASHPAGE_ERASE_STATE, sizeof(buf_empty));
+    size_t len = MAX_SIZE_STORAGE + sizeof(buf_empty);
+    uint8_t buf_read[len];
```

Comment:
This represent an static array so, could be the same that 
```suggestion
    uint8_t buf_read[MAX_SIZE_STORAGE +  3]; // Represent sizeof buff_empty array //
    /* Option 2 */
    uint8_t buf_read[19];
```